### PR TITLE
Fixed casting warnings on linux and macOS

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

File: wpad_dhcp_posix.c [hunk 1]
```diff
@@ -168,15 +169,15 @@ static bool dhcp_send_inform(SOCKET sfd, uint32_t xid, net_adapter_s *adapter) {
     opts = dhcp_copy_option(opts, &opt_end);
 
     // Broadcast DHCP request
-    int32_t request_len = (int32_t)(opts - (uint8_t *)&request);
-    int sent = sendto(sfd, (const char *)&request, request_len, 0, (struct sockaddr *)&address, sizeof(address));
+    ssize_t request_len = (ssize_t)(opts - (uint8_t *)&request);
+    ssize_t sent = sendto(sfd, (const char *)&request, request_len, 0, (struct sockaddr *)&address, sizeof(address));
```
Comment (sergio-nsk) [inline wpad_dhcp_posix.c:173]:
```suggestion
    const ssize_t request_len = (ssize_t)(opts - (uint8_t *)&request);
    const ssize_t sent = sendto(sfd, (const char *)&request, request_len, 0, (struct sockaddr *)&address, sizeof(address));
```

**Comment by sergio-nsk** [comment]
Disabled auto-merge if you want to follow the suggestions.

File: wpad_dhcp_posix.c [hunk 2]
```diff
@@ -168,15 +169,15 @@ static bool dhcp_send_inform(SOCKET sfd, uint32_t xid, net_adapter_s *adapter) {
...
     return sent == request_len;
 }
 
 static bool dhcp_read_reply(SOCKET sfd, uint32_t request_xid, dhcp_msg *reply) {
-    int response_len = recvfrom(sfd, (char *)reply, sizeof(dhcp_msg), 0, NULL, NULL);
+    ssize_t response_len = recvfrom(sfd, (char *)reply, sizeof(dhcp_msg), 0, NULL, NULL);
```
[the remark this answers is not in the document]
Comment (nmoinvaz) [reply wpad_dhcp_posix.c]: Fixed.

File: wpad_dhcp_posix.c (hunk 1 above)
Comment (nmoinvaz) [reply wpad_dhcp_posix.c]: Fixed.

**Pull request merged**